### PR TITLE
Feature/ope 327 voyagesmarketdata add advanced endpoint

### PR DESCRIPTION
**Pull request by lkolonis-signal**
Adding a function to call the advanced endpoint in voyages_market_data_api, added a post function in request_helpers, updated a Notebook example

### CODE REVIEW

**Review by edervisis — CHANGES_REQUESTED** [review]
Please update the Versions.txt and list the new version.md file in the mkdocs.yml

File: signal_ocean/voyages_market_data/voyages_market_data_api.py [hunk 1]
```diff
@@ -222,3 +225,130 @@ def get_voyage_market_data(
...
+            sources_include: If provided, market data from the give sources
+                will be included.
+            sources_exclude: If provided, market data from the give sources
+                will be excluded.
+
+        Returns:
+            A tuple containing the returned voyage market data.
+        """
+        data_body: Dict[str, Any] = {}
+        if imos:
```
[the remark this answers is not in the document]
Comment (dmouratidis-signal) [reply signal_ocean/voyages_market_data/voyages_market_data_api.py:307]: A workaround would be to retrieve function variables using `var()` function and then use `parsing_helpers._to_camel_case` function in order to convert variable names to camel case.

[the remark this answers is not in the document]
Comment (lkolonis-signal) [reply signal_ocean/voyages_market_data/voyages_market_data_api.py:307]: Thank you all, I ll check this out !

[the remark this answers is not in the document]
Comment (lkolonis-signal) [reply signal_ocean/voyages_market_data/voyages_market_data_api.py]: So, there were too many parameters that couldn't be converted by _to_camel_case so I added to the function the possibility to pass a dictionary of transformations. Then I created the body like :

```
        data_body: Dict[str, Any] = {}

        rename_keys = {'imos' : 'IMOs',
                'voyage_ids' : 'VoyageIDs',
                'vessel_class_ids' : 'VesselClassIDs',
                'trade_id' : 'TradeID',
                'charterer_ids_include' : 'ChartererIDsInclude',
                'charterer_ids_exclude' : 'ChartererIDsExclude',
                'cargo_type_ids_include' : 'CargoTypeIDsInclude',
                'cargo_type_ids_exclude' : 'CargoTypeIDsExclude'                
        }

        for key, value in locals().items():
            if key != 'self':
                if value:
                    cc_key = _to_camel_case(key, rename_keys=rename_keys)
                    data_body[cc_key] = value

        del data_body['DataBody']
        del data_body['RenameKeys']
```

[the remark this answers is not in the document]
Comment (lkolonis-signal) [reply signal_ocean/voyages_market_data/voyages_market_data_api.py]: Also, I have merged versioning info with the edits from Dimitris feature to produce 1 release, so v2.1.1 contains now info on both too

**Review by IsidorosTsalapatis — APPROVED** [review]

**Review by dmouratidis-signal — APPROVED** [review]

**Review by edervisis — APPROVED** [review]

**Pull request merged**